Podcast/UserSettings.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  UserSettings.swift
+//  Podcast
+//
+//  Created by Drew Dunne on 10/23/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+class UserSettings: NSObject {
+    
+    static let sharedSettings = UserSettings()
+    
+    private var saveCounter: Int!
+    private let maxSave: Int = 5
+    
+    var pushNotifications: Bool!
+    
+    private override init() {
+        saveCounter = 0
+        super.init()
+        loadData()
+    }
+    
+    private func loadData() {
+        pushNotifications = UserDefaults.standard.bool(forKey: "push_notifications")
+    }
+    
+    func writeToFile() {
+        UserDefaults.standard.set(pushNotifications, forKey: "push_notifications")
+    }
+    
+    static let changeUsernamePage: SettingsPageViewController = {
+        let changeUsernameVC = SettingsPageViewController()
+        let settings = [
+            SettingsSection(items: [
+                SettingsField(title: "New Username", saveFunction: { text in
+                    guard let username = text as? String else { return }
+                    let changeUsernameEndpointRequest = ChangeUsernameEndpointRequest(username: username)
+                    changeUsernameEndpointRequest.success = { (endpointRequest: EndpointRequest) in
+                        if let resp = endpointRequest.processedResponseValue as? Dictionary<String, Any> {
+                            System.currentUser = resp["user"] as? User
+                        }
+                        changeUsernameVC.navigationController?.popViewController(animated: true)
+                    }
+                    changeUsernameEndpointRequest.failure = { (endpointRequest: EndpointRequest) in
+                        //changeUsernameVC.setError(at: IndexPath(row: 0, section: 0), with: "Bad Username")
+                        // TODO: add error handling
+                    }
+                    System.endpointRequestQueue.addOperation(changeUsernameEndpointRequest)
+                }, type: .textField, placeholder: "new username")
```

Comment:
It might make more sense for the placeholder to be the current username, not sure if designs have come in for this yet. Also this placeholder text doesn't show up for me:
![image](https://user-images.githubusercontent.com/20246620/31976040-1329145a-b903-11e7-87b7-1c449dbb3148.png)

Reply:
Yeah I had a plan for that, I'm rethinking some stuff so this will change soon.